src/shm.c

```diff
@@ -286,11 +289,13 @@ shm_fini(uperf_shm_t *shm)
 void
 shm_update_strand_exit(uperf_shm_t *shm)
 {
-#ifdef HAVE_ATOMIC_H
+#ifdef HAVE_STDATOMIC_H
+	atomic_fetch_add(&shm->finished, 1);
+#elif HAVE_ATOMIC_H
```

Comment:
I don't believe `elif` works like `ifdef`.  It would seem you need to next them, see comment in `src/execute.c`.

Reply:
Strike that.  At least under the environment available to me with Fedora 34 it works as coded.

You might want to consider using the nested `ifdef` regardless to work under environments which don't support that behavior of `elif`.

Reply:
Your comment lead me to look into it more (autoconf learning, w00t). Apparently while it does work  the `elif` should be written as `#elif defined(HAVE_STDATOMIC_H)` since there could be some situation when is HAVE_ATOMIC_H is defined but not at to the correct value. Since autoconf use `#define HAVE_ATOMIC_H  1` and `#undefine HAVE_ATOMIC_H` when it is not present, the code works. I am going to update to include `#elif defined(HAVE_STDATOMIC_H)`  so we can be sure.

Reply:
I think @portante meant "nest", but I believe that he's correct that `#elif` works like `#if` and _not_ like `#ifdef`.

Is the `defined()` "directive" available?  I.e.,
```
#ifdef HAVE_STDATOMIC_H
	atomic_fetch_add(&shm->finished, 1);
#elif defined(HAVE_ATOMIC_H)
```
or, more symmetrically,
```
#if defined(HAVE_STDATOMIC_H)
	atomic_fetch_add(&shm->finished, 1);
#elif defined(HAVE_ATOMIC_H)
```

Reply:
> Is the defined() "directive" available?

Why, yes, it is....  (Now that I've looked at the diffs....)